addons/invtracker/invtracker.lua

```diff
@@ -0,0 +1,460 @@
+    update_bag(settings.SlotImage.mogWardrobe,items.wardrobe3,items.max_wardrobe3,items.enabled_wardrobe3)
+    update_bag(settings.SlotImage.mogWardrobe,items.wardrobe4,items.max_wardrobe4,items.enabled_wardrobe4)
+    update_temp_bag(settings.SlotImage.tempInventory,items.treasure,#items.treasure,items.enabled_treasure)
+    --update_temp_bag(settings.SlotImage.tempInventory,items.temporary,items.max_temporary,items.enabled_temporary)
+end
+
+function update_bag(config, bag, max, enabled)
+    if (config.visible and enabled) then
+        initialize_block()
+        print_bag(config, bag, max)
+    end
+end
+
+function update_temp_bag(config, bag, max)
+    if (config.visible and bag.enabled) then
+        initialize_block()
+        for key=1,max,1 do
+            print_item(config,-1,max)
+        end
+    end
+end
+
+function initialize_block()
+    current_block = current_block + 1
+    if slot_images[current_block] == nil then
+        slot_images[current_block] = {}
+    end
+    current_slot = 1
+    current_row = 1
+    current_column = 1
+end
+
+function print_equipment(status)
+  if (status > 0) then
+      print_slot(settings.SlotImage.status.equipment,settings.SlotImage.equipment.maxColumns,16)
+  else
+      print_slot(settings.SlotImage.status.empty,settings.SlotImage.equipment.maxColumns,16)
+  end
+end
+
+function print_bag(config, bag, max)
+    table.sort(bag, function(a,b)
+        if (a.status ~= b.status) then
+            return a.status > b.status
+        end
+        return a.count > b.count
+    end)
+    for key=1,max,1 do
+      if (bag[key].count > 0) then
+          print_item(config,bag[key].status,max)
+      else
+          print_slot(settings.SlotImage.status.empty,config.maxColumns,max)
+      end
+    end
+end
+
+function print_item(config, status, last_index)
+    if (status == -1) then
+        print_slot(settings.SlotImage.status.tempItem,config.maxColumns,last_index)
+    elseif (status == 0) then
+        print_slot(settings.SlotImage.status.default,config.maxColumns,last_index)
+    elseif (status == 5) then
+        print_slot(settings.SlotImage.status.equipped,config.maxColumns,last_index)
+    elseif (status == 19) then
+        print_slot(settings.SlotImage.status.linkshell_equipped,config.maxColumns,last_index)
+    elseif (status == 25) then
+        print_slot(settings.SlotImage.status.bazaar,config.maxColumns,last_index)
+    else
+        print_slot(settings.SlotImage.status.empty,config.maxColumns,last_index)
+    end
+end
+
+function print_slot(status, max_columns, last_index)
+    update_coordinates()
+    if slot_images[current_block][current_slot] == nil then
+        slot_images[current_block][current_slot] = {}
+    end
+    print_slot_background(status.background.color)
+    print_slot_box(status.color)
+    update_indexes(max_columns,last_index)
+end
+
+function print_slot_background(slot_color, max_columns, last_index)
```

Comment:
Might be worth introducing a local variable for `slot_images[current_block][current_slot]` to reduce the clutter. Same below in `print_slot_box`.